### PR TITLE
Bulletin Board: Timestamp and limit size on login

### PR DESCRIPTION
**Pull request by RoboMWM**
### Timestamped bulletin board messages
- All bulletin board messages added from this version on will be internally timestamped.
- Old bulletin board messages will continue to work as per usual
- Timestamp is displayed using a hoverable TextComponent.
  - The Paper API was added to support this (since we're only using spigot APIs, this is also compatible with spigot)
  - Falls back to standard way of printing bb messages if no timestamp can be found or a TextComponent can't be created (i.e. using CB)

You may want to refactor some of the methods I added if you wish to do so (e.g. Clan#displayBb). I've kept the original methods mostly as-is to mitigate merge conflicts in case this is not pulled.

### Bulletin board size on login
- The config key `bb.login-size` is used, and defaults to the value in `size` if not set.

I have not added this key to the default config.yml

If you're interested in pulling this, I'll be happy to refactor and add the config node, unless if you want to do that.

### CODE REVIEW

**Comment by RoboMWM** [comment]
Re: travis failure, it seems to not like Paper's SSL cert or something? Not sure what the deal is, it works fine for me https://travis-ci.org/MLG-Fortress/MountainDewritoes/

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/Dates.java [hunk 1]
```diff
@@ -135,4 +135,61 @@ private static long getTimeInMilliseconds(Date date) {
...
     }
+
+    /**
+     * Returns the string representation of the amount of time labeled as days, hours, minutes, seconds
+     * @author RoboMWM, LaxWasHere
+     * @param seconds the time in seconds
+     * @param depth Max amount of detail (e.g. only display days and hours if set to 1 and seconds > 1 day)
+     * @return
+     */
+    public static String formatTime(Long seconds, int depth)
```
Comment (RoinujNosde) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/Dates.java]: This method returns a hardcoded String.

**Comment by RoboMWM** [comment]
Did somebody comment here or something and then delete? Got a notification for this PR but I don't see anything new here.

**Comment by RoinujNosde** [comment]
@RoboMWM that was me. I was going to open a review, but changed my mind

**Comment by RoboMWM** [comment]
@RoinujNosde are you thinking of pulling this at all? Just wondering if I should continue to maintain my fork or if I should continue to make PRs here.

**Comment by RoinujNosde** [comment]
> I'll be happy to refactor and add the config node,

That would be great

**Comment by RoboMWM** [comment]
Ok..

**Comment by RoboMWM** [comment]
Oops, forgot about this. Nice to see it pulled though. Where did you resolve the merge conflict? Usually it should show up in the PR but I guess it didn't here.

**Comment by RoinujNosde** [comment]
> Where did you resolve the merge conflict?

9b69020

> It was because travis isn't using Java 8. Also, I might've forgotten that spigot requires player#spigot()#sendMessage to send a message with BaseComponents (apparently Paper just folds it in with player#sendMessage), though the try-catch will just have it fall back to default on spigot. If there's any interest in pulling this PR (since there's been no feedback thus far), I'll update this (via doing the above) to give pretty hoverable messages on spigot too.

I skipped this message when merging...

**Comment by RoboMWM** [comment]
Oh I see, you did the merge conflict during the pull instead of on my branch. You may also want to consider using the squash commits option when merging PRs so a) you don't get all the "typo" commits from the PR and b) the commit history will better reflect when changes occurred in this repo.

**Pull request merged**